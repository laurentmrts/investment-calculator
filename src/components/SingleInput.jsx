export default function SingleInput({label, inputsToCalculate, onChange}) {
  function addSpacesBeforeUppercaseToUppercase(str) {
    return str.replace(/([A-Z])/g, ' $1').trim().toUpperCase();
  }

  return (
    <p>
      <label>{addSpacesBeforeUppercaseToUppercase(label)}</label>
      <input
        type="number"
        required
        value={inputsToCalculate[label]}
        onChange={(event) =>
          onChange(label, event.target.value)
        }
      />
    </p>
  );
}