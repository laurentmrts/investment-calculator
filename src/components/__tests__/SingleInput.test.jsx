import {describe, it, beforeEach, expect, vi} from "vitest";
import {render, screen} from "@testing-library/react";
import SingleInput from "../SingleInput.jsx";
import {userEvent} from "@testing-library/user-event";

describe('SingleInput component', () => {

    let onChange;

    beforeEach(() => {
        onChange = vi.fn()
        render(<SingleInput label="aCamelCaseLabel" inputsToCalculate={1} onChange={onChange} testId="testId"/>)
    })

    it('should match snapshot', async () => {
        expect(await screen.findByTestId('testId')).toMatchSnapshot()
    })

    it('displays upper cased label', async () => {
        expect(await screen.findByTestId('testId-label')).toHaveTextContent('A CAMEL CASE LABEL')
    })

    it('has an input that trigger onChange', async () => {
        const input = await screen.findByTestId('testId-input')
        expect(input).toHaveValue(1)
        await userEvent.type(input, '2')
        expect(onChange).toHaveBeenCalledTimes(1)
        expect(onChange).toHaveBeenCalledWith('aCamelCaseLabel', '12')
    })

})
