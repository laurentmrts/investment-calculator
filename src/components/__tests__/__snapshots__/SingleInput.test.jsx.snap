// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SingleInput component > should match snapshot 1`] = `
<p
  data-testid="testId"
>
  <label
    data-testid="testId-label"
  >
    A CAMEL CASE LABEL
  </label>
  <input
    data-testid="testId-input"
    required=""
    type="number"
    value="1"
  />
</p>
`;
